config/__tests__/index.test.js

```diff
@@ -9,21 +9,18 @@ const { loadHashiConfigForEnvironment } = require('../index')
 const fixtureDir = path.join(__dirname, '__fixtures__', 'config-loading')
 
 describe('loadHashiConfigByEnvironment', () => {
-	let cwd = process.cwd()
-
 	beforeAll(() => {
 		process.env.HASHI_ENV = 'production'
-		process.chdir(fixtureDir)
 	})
 
 	afterAll(() => {
-		process.chdir(cwd)
 		process.env.HASHI_ENV = undefined
 	})
 
 	test('loads configuration and handles extending', () => {
+		vi.spyOn(process, 'cwd').mockReturnValue(fixtureDir)
```

Comment:
Instead of _actually_ changing directories (which isn't supported in vitest), we simply mock the `process.cwd()` call to return the `fixtureDir`.